### DISCUSSION TITLE
Ajax Modal

### DISCUSSION THREAD

**Discussion by aliqasemzadeh** [wireui/wireui#53] [Q&A]
Hi
I think we can have ajax modal

**Comment by ph7jack** [comment]
What do you think about modal ajax?
Describe more about "ajax modal"

**Comment by aliqasemzadeh** [comment]
@PH7-Jack 
Hi
For example load data from another component base on button click.

**Comment by aliqasemzadeh** [comment]
@PH7-Jack 
For Example:
````
        <x-modal blur fullscreen wire:model.defer="panelModal">
            <x-card fullscreen>
                <livewire:panel symbol="???" >
            </x-card>
        </x-modal>
````

While symbol will change base on `onclick="$openModal('panelModal')"`

**Reply by vlados** [reply]
> @PH7-Jack For Example:
> 
> ```
>         <x-modal blur fullscreen wire:model.defer="panelModal">
>             <x-card fullscreen>
>                 <livewire:panel symbol="???" >
>             </x-card>
>         </x-modal>
> ```
> 
> While symbol will change base on `onclick="$openModal('panelModal')"`

You can achieve that with simple events
1. onclick fire event loadpanel(id)
2. panel listen for event loadpanel
Done :)

**Comment by ph7jack** [comment]
When modal is opening, the livewire component receives a value?

**Comment by aliqasemzadeh** [comment]
@PH7-Jack 
Yes exactlly.

**Comment by aliqasemzadeh** [comment]
@PH7-Jack 
Can you guide me how to do it?

**Comment by ph7jack** [comment]
I believe this cannot be done, the variables passed in the livewire component are not reactive. But it is possible to do this using livewire events.

Create a listener into your component to receives and set the new value
Before opening a modal, send an event with a new value to set

Ex:
```
Outside component
public function openModal() {
    $this->modal = true;
    $this->emit('set-value', $newValue);
}

Component Inside Modal
protected $listeners =['set-value' => 'setValue'];

public function setValue($value) {
    $this->myVar = $value;
}
```

**Comment by trippo** [comment]
I think this can be closed

**Comment by ph7jack** [comment]
I will go move it to one discussion, if the events solve the problem, I will mark it as complete
